### PR TITLE
ci/tests: cleaning standalone script

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

simplify & unify standalone scripts for maintenance

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--19141.org.readthedocs.build/en/19141/

cc @carmocca @borda @justusschock @awaelchli

### CODE REVIEW

File: tests/tests_pytorch/run_standalone_tasks.sh [hunk 1]
```diff
@@ -18,11 +18,16 @@ set -e
 # this environment variable allows special tests to run
 export PL_RUN_STANDALONE_TESTS=1
 
-can_run_nvprof=$(python -c "import torch; print(torch.cuda.is_available() and torch.cuda.get_device_capability()[0] < 8)")
-if [[ $can_run_nvprof == "True" ]]; then
-    echo "Running profilers/test_profiler.py::test_pytorch_profiler_nested_emit_nvtx"
-    nvprof --profile-from-start off -o trace_name.prof -- python -m coverage run --source lightning.pytorch --append -m pytest --no-header profilers/test_profiler.py::test_pytorch_profiler_nested_emit_nvtx
-fi
+#can_run_nvprof=$(python -c "import torch; print(torch.cuda.is_available() and torch.cuda.get_device_capability()[0] < 8)")
```
Comment (Borda) [inline tests/tests_pytorch/run_standalone_tasks.sh:21]: note this was not run in any our CI

File: pyproject.toml [hunk 2]
```diff
@@ -269,7 +269,8 @@ addopts = [
     "--ignore=legacy/checkpoints",
 ]
 markers = [
-    "cloud:Run the cloud tests for example",
+    "cloud: Run the cloud tests for example",
+    "standalone: Run the test in single pytest execution",
```
Comment (Borda) [inline pyproject.toml:273]:
```suggestion
```

File: tests/tests_fabric/run_tpu_tests.sh [hunk 3]
```diff
@@ -30,12 +30,12 @@ python3 -c "from lightning.fabric.accelerators.xla import _XLA_AVAILABLE; print(
 python3 -c "from lightning.fabric.accelerators import XLAAccelerator; assert XLAAccelerator.is_available()"
 echo "Sanity check passed!"
 
-echo "--- Running Fabric tests ---"
+echo "--- Running Regular tests ---"
```
Comment (Borda) [inline tests/tests_fabric/run_tpu_tests.sh:33]:
```suggestion
echo "--- Running Fabric tests ---"
```

File: tests/tests_pytorch/conftest.py [hunk 4]
```diff
@@ -159,8 +159,7 @@ def thread_police_duuu_daaa_duuu_daaa():
             continue
         elif thread.name == "fsspecIO":
             continue
-        else:
-            raise AssertionError(f"Test left zombie thread: {thread}")
+        raise AssertionError(f"Test left zombie thread: {thread}")
```
Comment (Borda) [inline tests/tests_pytorch/conftest.py:162]:
```suggestion
        else:
            raise AssertionError(f"Test left zombie thread: {thread}")
```

File: tests/tests_pytorch/profilers/test_profiler.py [hunk 5]
```diff
@@ -471,20 +471,21 @@ def look_for_trace(trace_dir):
...
-    trainer = Trainer(
-        fast_dev_run=True,
-        profiler=profiler,
-        accelerator="gpu",
-        devices=1,
-        enable_progress_bar=False,
-        enable_model_summary=False,
-    )
-    trainer.fit(model)
+# @RunIf(min_cuda_gpus=1, standalone=True)
```
Comment (ethanwharris) [inline tests/tests_pytorch/profilers/test_profiler.py:474]: Maybe add a TODO or some comment explaining:
- why it is commented out
- why it isn't just deleted

Comment (Borda) [reply tests/tests_pytorch/profilers/test_profiler.py:474]:
```suggestion
# Todo: this test has not been running as all our CI GPU runners have higher capacity
# @RunIf(min_cuda_gpus=1, standalone=True)
```
